sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,541 @@
+        while not rospy.is_shutdown():
+            self.run()
+            self.command.reset()
+
+    def run(self):
+        self.print_help()
+        in_string = raw_input("Please enter command:").upper()
+        if not self.command.parse(in_string):
+            return
+        if self.command.quit():
+            rospy.loginfo("Quitting...")
+            sys.exit(0)
+        rospy.loginfo("type: %s joint: %s finger: %s angle: %s", self.command.cmd, self.command.joint,
+                      self.command.finger, self.command.joint)
+        rospy.loginfo("\n\n\n\n\n")
+        if not self.command.all:
+            if self.command.position_mode():
+                self.test_joint(self.command.finger_joint, mode='position', value=self.command.value)
+            if self.command.effort_mode():
+                file_prefix = raw_input("Name/describe the test (adds text to filename, optional)")
+                self.test_joint(self.command.finger_joint, mode='PWM', value=str(float(self.command.value)),
+                                prefix=file_prefix + '_plus')
+                rospy.sleep(1)
+                self.test_joint(self.command.finger_joint, mode='PWM', value=str(float(self.command.value)*-1.0),
+                                prefix=file_prefix + '_minus')
+                rospy.sleep(1)
+        else:
+            for joint in self.command.joints_list:
+                rospy.logwarn("acting on: %s", joint)
+            if self.command.position_mode():
+                file_prefix = raw_input("enter file prefix:")
+                for joint in self.command.joints_list:
+                    if file_prefix == '':
+                        file_prefix = None
+                    self.test_joint(joint, mode='testing', prefix=file_prefix)
+            elif self.command.effort_mode():
+                file_prefix = raw_input("enter file prefix:")
+                for joint in self.command.joints_list:
+                    if 'LF' in joint.upper() or 'RF' in joint.upper():
+                        self.command.value = str(float(self.command.value)*(-1.0))
+                    self.test_joint(joint, 'PWM', value=self.command.value, prefix=file_prefix + '_plus')
+                    rospy.sleep(1)
+                    self.test_joint(joint, 'PWM', value=str(float(self.command.value)*(-1.0)),
+                                    prefix=file_prefix + '_minus')
+                    rospy.sleep(1)
+
+    def switch_finger_to_effort(self, finger):
+        joints_to_change = []
+        for joint in self.requested_joints:
+            if finger in joint:
+                joints_to_change.append(joint)
+        temp_controller_helper = ControllerHelper([self._hand_prefix[0] + 'h'], [self._hand_prefix],
+                                                  [joint.lower() for joint in joints_to_change])
+        temp_controller_helper.change_hand_ctrl("effort")
+
+    def switch_finger_to_position(self, finger):
```

Comment:
this is literally the same code as about apart from the positon and effort words. Please use shared method with parameter